```
In static member function ‘static _Up* std::__copy_move<_IsMove, true, std::random_access_iterator_tag>::__copy_m(_Tp*, _Tp*, _Up*) [with _Tp = short int; _Up = short int; bool _IsMove = false]’,
    inlined from ‘_OI std::__copy_move_a2(_II, _II, _OI) [with bool _IsMove = false; _II = short int*; _OI = short int*]’ at /opt/rh/gcc-toolset-13/root/usr/include/c++/13/bits/stl_algobase.h:506:30,
    inlined from ‘_OI std::__copy_move_a1(_II, _II, _OI) [with bool _IsMove = false; _II = short int*; _OI = short int*]’ at /opt/rh/gcc-toolset-13/root/usr/include/c++/13/bits/stl_algobase.h:533:42,
    inlined from ‘_OI std::__copy_move_a(_II, _II, _OI) [with bool _IsMove = false; _II = short int*; _OI = short int*]’ at /opt/rh/gcc-toolset-13/root/usr/include/c++/13/bits/stl_algobase.h:540:31,
    inlined from ‘_OI std::copy(_II, _II, _OI) [with _II = short int*; _OI = short int*]’ at /opt/rh/gcc-toolset-13/root/usr/include/c++/13/bits/stl_algobase.h:633:7,
    inlined from ‘static void TAO::details::value_traits<T, dummy>::copy_range(value_type*, value_type*, iter) [with iter = short int*; T = short int; bool dummy = true]’ at /sandbox/gitlab-runner-ibst/builds/ibst/products/open-source/OpenDDW/spm-rhel9-gcc13/open-source/opendds/include/tao/Value_Traits_T.h:52:14,
    inlined from ‘static void TAO::details::value_traits<T, dummy>::copy_swap_range(value_type*, value_type*, iter) [with iter = short int*; T = short int; bool dummy = true]’ at /sandbox/gitlab-runner-ibst/builds/ibst/products/open-source/OpenDDW/spm-rhel9-gcc13/open-source/opendds/include/tao/Value_Traits_T.h:60:15,
    inlined from ‘void TAO::details::generic_sequence<T, ALLOCATION_TRAITS, ELEMENT_TRAITS>::length(CORBA::ULong) [with T = short int; ALLOCATION_TRAITS = TAO::details::unbounded_value_allocation_traits<short int, true>; ELEMENT_TRAITS = TAO::details::value_traits<short int, true>]’ at /sandbox/gitlab-runner-ibst/builds/ibst/products/open-source/OpenDDW/spm-rhel9-gcc13/open-source/opendds/include/tao/Generic_Sequence_T.h:239:36,
    inlined from ‘void TAO::details::generic_sequence<T, ALLOCATION_TRAITS, ELEMENT_TRAITS>::length(CORBA::ULong) [with T = short int; ALLOCATION_TRAITS = TAO::details::unbounded_value_allocation_traits<short int, true>; ELEMENT_TRAITS = TAO::details::value_traits<short int, true>]’ at /sandbox/gitlab-runner-ibst/builds/ibst/products/open-source/OpenDDW/spm-rhel9-gcc13/open-source/opendds/include/tao/Generic_Sequence_T.h:198:8,
    inlined from ‘void TAO::unbounded_value_sequence<T, Tag>::length(CORBA::ULong) [with T = short int; Tag = int]’ at /sandbox/gitlab-runner-ibst/builds/ibst/products/open-source/OpenDDW/spm-rhel9-gcc13/open-source/opendds/include/tao/Unbounded_Value_Sequence_T.h:60:17,
    inlined from ‘void {anonymous}::init(DDS::TopicQos&)’ at /sandbox/gitlab-runner-ibst/builds/ibst/products/open-source/OpenDDW/OpenDDW/src/qos_dictionary.cpp:54:40,
    inlined from ‘DDS::TopicQos QosDictionary::Topic::bestEffort()’ at /sandbox/gitlab-runner-ibst/builds/ibst/products/open-source/OpenDDW/OpenDDW/src/qos_dictionary.cpp:228:9:
/opt/rh/gcc-toolset-13/root/usr/include/c++/13/bits/stl_algobase.h:437:30: error: ‘void* __builtin_memmove(void*, const void*, long unsigned int)’ forming offset 2 is out of the bounds [0, 2] [-Werror=array-bounds=]
  437 |             __builtin_memmove(__result, __first, sizeof(_Tp) * _Num);
      |             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/src/qos_dictionary.cpp
+++ b/src/qos_dictionary.cpp
@@ -15,7 +15,7 @@
 #include <string>
 
 //gcc13 warnings
-#if defined(__GNUC__)
+#if __GNUC__ && defined( __has_warning )
 #  pragma GCC diagnostic ignored "-Warray-bounds"
 #  pragma GCC diagnostic ignored "-Wstringop-overflow"
 #endif

--- a/src/qos_dictionary.cpp
+++ b/src/qos_dictionary.cpp
@@ -1,12 +1,12 @@
 #ifdef WIN32
-#pragma warning(push, 0)  //No DDS warnings
+#  pragma warning(push, 0)  //No DDS warnings
 #endif
 
 #include <dds/DdsDcpsInfrastructureC.h>
 #include <dds/DCPS/Marked_Default_Qos.h>
 
 #ifdef WIN32
-#pragma warning(pop)
+#  pragma warning(pop)
 #endif
 
 #include "qos_dictionary.h"
@@ -16,8 +16,12 @@
 
 //gcc13 warnings
 #if __GNUC__ && defined( __has_warning )
-#  pragma GCC diagnostic ignored "-Warray-bounds"
-#  pragma GCC diagnostic ignored "-Wstringop-overflow"
+#  if __has_warning( "-Warray-bounds" )
+#    pragma GCC diagnostic ignored "-Warray-bounds"
+#  endif
+#  if __has_warning( "-Wstringop-overflow" )
+#      pragma GCC diagnostic ignored "-Wstringop-overflow"
+#  endif
 #endif
 
 namespace

--- a/src/qos_dictionary.cpp
+++ b/src/qos_dictionary.cpp
@@ -15,7 +15,7 @@
 #include <string>
 
 //gcc13 warnings
-#ifndef WIN32
+#if defined(__GNUC__)
 #  pragma GCC diagnostic ignored "-Warray-bounds"
 #  pragma GCC diagnostic ignored "-Wstringop-overflow"
 #endif

